collector/metrics.go

```diff
@@ -0,0 +1,241 @@
+
+	// Exporter health
+	Up *prometheus.Desc
+}
+
+// NewMetricDescriptors creates and returns all metric descriptors for the Databricks exporter.
+func NewMetricDescriptors() *MetricDescriptors {
+	return &MetricDescriptors{
+		// ===== Billing & Cost Metrics (FinOps) =====
+
+		BillingDBUsTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "billing", "dbus_total"),
+			"Daily Databricks Units (DBUs) consumed per workspace and SKU.",
+			[]string{labelWorkspaceID, labelSKUName},
+			nil,
+		),
+
+		BillingCostEstimateUSD: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "billing", "cost_estimate_usd"),
+			"Daily list-price cost estimate (DBUs × list price) per workspace and SKU.",
+			[]string{labelWorkspaceID, labelSKUName},
+			nil,
+		),
+
+		PriceChangeEvents: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "price_change_events"),
+			"Count of pricing changes for a SKU from historical list prices.",
+			[]string{labelSKUName},
+			nil,
+		),
+
+		BillingExportErrorsTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "billing", "export_errors_total"),
+			"Exporter error count segmented by stage (sql, publish, etc.).",
+			[]string{labelStage},
+			nil,
+		),
+
+		BillingCSVRowsIngestedTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "billing", "csv_rows_ingested_total"),
+			"Rows ingested from Billable Usage CSV (fallback path) for pipeline health.",
+			nil,
+			nil,
+		),
+
+		// ===== Jobs Metrics (SRE/Platform) =====
+
+		JobRunsTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "job_runs_total"),
+			"Number of Lakeflow Jobs runs per workspace, job ID and job name.",
+			[]string{labelWorkspaceID, labelJobID, labelJobName},
+			nil,
+		),
+
+		JobRunStatusTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "job_run_status_total"),
+			"Job status counts (SUCCEEDED/FAILED/CANCELED) per workspace, job ID and job name.",
+			[]string{labelWorkspaceID, labelJobID, labelJobName, labelStatus},
+			nil,
+		),
+
+		JobRunDurationSeconds: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "job_run_duration_seconds"),
+			"Job run duration quantiles (p50/p95/p99) per workspace, job ID and job name.",
+			[]string{labelWorkspaceID, labelJobID, labelJobName, labelQuantile},
+			nil,
+		),
+
+		TaskRetriesTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "task_retries_total"),
+			"Number of retries across job tasks per workspace, job ID, job name and task key.",
+			[]string{labelWorkspaceID, labelJobID, labelJobName, labelTaskKey},
+			nil,
+		),
+
+		JobSLAMissTotal: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "", "job_sla_miss_total"),
+			"Job runs exceeding configured SLA threshold per workspace, job ID and job name.",
+			[]string{labelWorkspaceID, labelJobID, labelJobName},
+			nil,
+		),
+
+		// ===== Pipelines Metrics (SRE/Platform) =====
+
+		PipelineRunsTotal: prometheus.NewDesc(
```

Comment:
Same as we discussed for `job_name` and `job_id`. Would be good to have a conditional here to let customers pick which to include, and perhaps default to only one of the two?

Reply:
As discussed on Slack, I opted in to keep both `job_id` and `job_name` since they have a 1:1 mapping, same goes for `pipeline_id` and `pipeline_name`. However, you were right to call out `task_id`, it definitely has a high cardinality since a single job can have many tasks and the mapping is 1:M.

(I also replied to your question about our Slack convo in the other comment about this.)